rllib/core/models/catalog.py

```diff
@@ -0,0 +1,227 @@
+import gymnasium as gym
+from ray.rllib.core.models.configs import (
+    MLPEncoderConfig,
+    LSTMEncoderConfig,
+    CNNEncoderConfig,
+)
+from ray.rllib.core.models.base import ModelConfig
+from ray.rllib.models import MODEL_DEFAULTS
+from gymnasium.spaces import Box
+from ray.rllib.models.utils import get_filter_config
+
+
+class Catalog:
+    """Defines how what models an RLModules builds.
+
+    RLlib's native RLModules get their Models from a Catalog object.
+    By default, that Catalog builds the configs it holds.
+    You can modify a Catalog so that it builds different Models by subclassing and
+    overriding the build_* methods. Alternatively, you can customize the configs
+    inside RLlib's Catalogs to customize what is being built by RLlib.
+
+    Usage example:
+
+    # Define a custom catalog
+
+    .. testcode::
+
+    class MyCatalog(Catalog):
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        model_config_dict: dict,
+    ):
+        super().__init__(observation_space, action_space, model_config_dict)
+        self.my_model_config = MLPModelConfig(
+            hidden_layer_dims=[64, 32],
+            input_dim=self.observation_space.shape[0],
+            output_dim=1,
+        )
+
+        def build_my_head(self, framework: str):
```

Comment:
is this meant to be indented into the scope of the __init__ function? If not pls unindent for clarity/correctness :)

Reply:
Thanks!